### PR TITLE
Complete DuckDNS add-on documentation

### PR DESCRIPTION
**Pull request by dnguyen800**
In reference to Issue #719.

I have added information to the "How to Use", "Configuration", and "Known Issues and Limitations" sections. Please let me know if you see any inconsistencies or discrepancies.

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @dnguyen800, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;hassio-addons#749).

Once you do that we will be able to review and accept this pull request.

Thanks!

File: duckdns/README.md [hunk 1]
```diff
@@ -18,6 +18,13 @@ The installation of this add-on is straightforward and easy to do.
 
 ## How to use
 
+1. Visit [DuckDNS.org](https://www.duckdns.org/) and create an account by logging in through any of the available account services (Google, Github, Twitter, Persona).
+2. In the `Domains` section, type the name of the sub domain you wish to register and click `add domain`.
```
Comment (frenck) [inline duckdns/README.md]:
```suggestion
2. In the `Domains` section, type the name of the subdomain you wish to register and click `add domain`.
```

File: duckdns/README.md [hunk 2]
```diff
@@ -18,6 +18,13 @@ The installation of this add-on is straightforward and easy to do.
...
+3. If registration was a success, the sub domain is listed in the `Domains` section along with `current ip` being the public IP address of the device you are currently using to access duckdns.org. The IP address will be updated by the DuckDNS add-on.
```
Comment (frenck) [inline duckdns/README.md]:
```suggestion
3. If registration was a success, the subdomain is listed in the `Domains` section along with `current ip` being the public IP address of the device you are currently using to access `duckdns.org`. The IP address will be updated by the DuckDNS add-on.
```

File: duckdns/README.md [hunk 3]
```diff
@@ -18,6 +18,13 @@ The installation of this add-on is straightforward and easy to do.
...
+4. In the DuckDNS add-on configuration, perform the following:
+    - Copy the DuckDNS token (listed at the top of the page where account details are displayed) from `duckdns.org` and paste into the `token` option.
+    - Update the `domains` option with the full domain name you registered. E.g. "my-domain.duckdns.org".
```
Comment (frenck) [inline duckdns/README.md]:
```suggestion
    - Update the `domains` option with the full domain name you registered. E.g., `my-domain.duckdns.org`.
```

File: duckdns/README.md [hunk 4]
```diff
@@ -36,20 +43,49 @@ Add-on configuration:
 }
 ```
 
-### Option: `lets_encrypt`
+### Option group `lets_encrypt`
+
+---
```
Comment (frenck) [inline duckdns/README.md]: Let us remove the line separator, there is no begin/end of this section.

File: duckdns/README.md [hunk 5]
```diff
@@ -36,20 +43,49 @@ Add-on configuration:
...
+
+The following options are for the option group: `lets_encrypt`. These settings
+only apply to Let's Encrypt SSL  certificates.
+
+
+#### Option `lets_encrypt`: `accept_terms`
```
Comment (frenck) [inline duckdns/README.md]: Let us make it consistent with how we handle services on our documentation website.

```suggestion
### Option `lets_encrypt.accept_terms`
```

File: duckdns/README.md [hunk 6]
```diff
@@ -36,20 +43,49 @@ Add-on configuration:
...
+
+Once you have read and accepted the Let's Encrypt [Subscriber Agreement](https://letsencrypt.org/repository/), change value to `true` in order to use Let's Encrypt services.
+
+
+#### Option `lets_encrypt`: `certfile`
```
Comment (frenck) [inline duckdns/README.md]:
```suggestion
### Option `lets_encrypt.certfile`
```

File: duckdns/README.md [hunk 7]
```diff
@@ -36,20 +43,49 @@ Add-on configuration:
...
+
+The name of the certificate file generated by Let's Encrypt. The file is used for SSL by Hass.io add-ons and is recommended to keep the filename as-is (`fullchain.pem`) for compatibility.
+
+**Note**: _The file is stored in `/ssl/`, which is the default for Hass.io_
+
+#### Option `lets_encrypt`: `keyfile`
```
Comment (frenck) [inline duckdns/README.md]:
```suggestion
#### Option `lets_encrypt.keyfile`
```

File: duckdns/README.md [hunk 8]
```diff
@@ -36,20 +43,49 @@ Add-on configuration:
...
+**Note**: _The file is stored in `/ssl/`, which is the default for Hass.io_
 
 
 ### Option: `token`
 
+The DuckDNS authentication token found at the top of the DuckDNS account landing page. The token is required to make any changes to the subdomains registered to your account.
 
 ### Option: `domains`
 
+A list of DuckDNS domains registered under your account. An acceptable naming convention is "my-domain.duckdns.org".
```
Comment (frenck) [inline duckdns/README.md]:
```suggestion
A list of DuckDNS subdomains registered under your account. An acceptable naming convention is `my-domain.duckdns.org`.
```

File: duckdns/README.md [hunk 9]
```diff
@@ -36,20 +43,49 @@ Add-on configuration:
...
 
 ### Option: `seconds`
 
+The number of seconds to wait before updating DuckDNS subdomains and renewing Let's Encrypt certificates.
 
 ## Known issues and limitations
 
+- To log in, DuckDNS requires a free account from any of the following services: Google, Github, Twitter, or Persona.
+- A free DuckDNS account is limited to five sub domains.
+
```
Comment (frenck) [inline duckdns/README.md]: Unneeded newline

```suggestion
```

File: duckdns/README.md [hunk 10]
```diff
@@ -36,20 +43,49 @@ Add-on configuration:
...
+A list of DuckDNS domains registered under your account. An acceptable naming convention is "my-domain.duckdns.org".
 
 ### Option: `seconds`
 
+The number of seconds to wait before updating DuckDNS subdomains and renewing Let's Encrypt certificates.
 
 ## Known issues and limitations
 
+- To log in, DuckDNS requires a free account from any of the following services: Google, Github, Twitter, or Persona.
+- A free DuckDNS account is limited to five sub domains.
```
Comment (frenck) [inline duckdns/README.md]:
```suggestion
- A free DuckDNS account is limited to five subdomains.
```

File: nginx_proxy/README.md [hunk 11]
```diff
@@ -0,0 +1,113 @@
+# Hass.io Core Add-on: NGINX Home Assistant SSL proxy
```
Comment (frenck) [inline nginx_proxy/README.md]: This should be a separate PR?

Comment (dnguyen800) [reply nginx_proxy/README.md]: Yes, sorry. I meant to submit as separate PR but still getting familiar with Github. I thought I reverted this change. Anyways, a separate PR should have been submitted anyways and this change can be canceled.
